lib/CL/pocl_llvm_build.cc

```diff
@@ -353,14 +355,45 @@ int pocl_llvm_build_program(cl_program program,
 
   POCL_MSG_PRINT_LLVM("all build options: %s\n", ss.str().c_str());
 
-  std::istream_iterator<std::string> begin(ss);
-  std::istream_iterator<std::string> end;
-  std::istream_iterator<std::string> i = begin;
-  std::vector<const char*> itemcstrs;
+  char replace_me = 0;
+
+  char *temp_options = (char *) malloc (ss.str().length() + 1);
+
+  memset (temp_options, 0, ss.str().length() + 1);
+  strncpy (temp_options, ss.str().c_str(), ss.str().length());
+
+  if (pocl_escape_quoted_whitespace (temp_options, &replace_me) == -1)
+  {
+    POCL_MEM_FREE (temp_options);
+    return CL_INVALID_BUILD_OPTIONS;
+  }
+
+  std::istringstream iss(temp_options);
+  std::vector<const char *> itemcstrs;
   std::vector<std::string> itemstrs;
-  while (i != end) {
-    itemstrs.push_back(*i);
-    ++i;
+
+  std::string s;
+
+  while (iss >> s)
+  {
+    // if needed, put back whitespace
+    if (replace_me != 0)
+    {
+      if (s.find(replace_me) != std::string::npos)
+      {
+        std::replace(s.begin(), s.end(), replace_me, ' ');
+      }
+    }
+
+    // if quoted, remove it to make compiler happy
```

Comment:
I'm not sure about this snippet here: Why do you do this here? Doesn't it break the tokenization when you have white spaces in the args for switches?

Reply:
Hi, I'm sure because I test it before push. If you don't strip out quotes, all includes declared in the kernel source are appended to the quoted string, consequently the compiler returns an error because it cannot find the file to include.

Example:

`
"/this/is your/path"/inc_vendor.h
`

Thanks

Reply:
Doesn't your code convert ```/foo/"bar dir"/inc_vendor.h````to ```/foo/ bar dir /inc_vendor.h```, that is, add extra whitespaces to the path?